File: grind75/Jan15/739.go
func dailyTemperatures(temperatures []int) []int {
    n := len(temperatures)
    ret := make([]int, n)
    if n == 1 { return ret }
    stack := []int{0}
    for i := 1; i < n; i++ {
        idx := stack[len(stack)-1]
        if temperatures[i] <= temperatures[idx] {
            stack = append(stack, i)
            continue
        }
        for len(stack) > 0 && temperatures[i] > temperatures[stack[len(stack)-1]]{
            pop := stack[len(stack)-1]
            stack = stack[:len(stack)-1]
            ret[pop] = i-pop
        }
        stack = append(stack, i)
    }
    return ret
}

File: grind75/Jan15/739.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> 
List[int]:
        n = len(temperatures)
        stack = []
        result = [0] * n
        for i, temp in enumerate(temperatures):
            while stack and stack[-1][1] < temp:
                result[stack[-1][0]] = i - stack[-1][0]
                stack.pop()
            stack.append([i, temp])
        return result
